File: Foody/app/src/main/java/prateek_gupta/foody/util/RecipesDiffUtils.java
package prateek_gupta.foody.util;

import androidx.recyclerview.widget.DiffUtil;

import java.util.List;

import prateek_gupta.foody.models.Result;

public class RecipesDiffUtils<T> extends DiffUtil.Callback {
    List<T> oldList;
    List<T> newList;

    public RecipesDiffUtils(List<T> oldList, List<T> newList) {
        this.oldList = oldList;
        this.newList = newList;
    }

    @Override
    public int getOldListSize() {
        return oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition)==newList.get(newItemPosition);
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition)==newList.get(newItemPosition);
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/util/RecipesDiffUtils.kt
package prateek_gupta.foody.util

import androidx.recyclerview.widget.DiffUtil
import prateek_gupta.foody.models.Result

class RecipesDiffUtils(
    private val oldList: List<Result>,
    private val newList: List<Result>
) :DiffUtil.Callback(){
    override fun getOldListSize(): Int {
        return oldList.size
    }

    override fun getNewListSize(): Int {
        return newList.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition]===newList[newItemPosition]
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition]==newList[newItemPosition]
    }
}